File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
answer = 0
def solution(numbers, target):
    def dfs(target,val,index):
        global answer
        index+=1
        if index == len(numbers):
            if val == target:
                answer+=1
            return
        dfs(target,val+numbers[index],index)
        dfs(target,val-numbers[index],index)
        
    dfs(target,numbers[0],0)
    dfs(target,-numbers[0],0)
    
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;
// import java.io.*

class Solution {
    int answer;
    void dfs(int index, int value, int target, int[] numbers,int len){
        
        if (index == len-1) {
            if (value == target){
               answer+=1; 
            }
            return;
        
        }
        dfs(index+1,value+numbers[index+1],target,numbers,len);
        dfs(index+1,value-numbers[index+1],target,numbers,len);
        
    }
    public int solution(int[] numbers, int target) {
        int index = 0;
        int len = numbers.length;
        dfs(index,numbers[index],target,numbers,len);
        dfs(index,-numbers[index],target,numbers,len);
        return answer;
    }
    
}